AprilSylph/Palettes-for-Tumblr
# Add new color variables to built-in palettes

### Description


The new Tumblr design system has a new (huge) set of theme color variables, which Palettes for Tumblr doesn't currently override (see #191). Still not sure what we want to do about user-generated palettes re: these new colors, as it seems impractical to ask palette authors to manually pick all of them (particularly when, from the outside, we have no clue what any of them are for).

This generates the color variables for the new Tumblr palette system for our built-in non-native palettes (2013, 2016, and decision) in-browser using CSS `color-mix` functions. They are chosen to match True Blue's palette system when run on the True Blue base palette colors.

This also copies the current set of variables into our native palette data and adds a nodejs/puppeteer script that can be run to keep the variable list up to date (contains the commits of #216)

### Testing steps


Select one of the built-in tumblr palettes that differs from the currently applied palette and confirm that elements such as the create button and notification badges are themed correctly. Confirm that elements are themed correctly on recently created pages that use the new theme color variables such as:

- https://www.tumblr.com/search/xkit/communities
- https://www.tumblr.com/communities/test-community-46174/settings
- https://www.tumblr.com/communities/communities-feedback/about

Repeat this with each of the built-in non-tumblr palettes.

One can press shift-P repeatedly to cycle the selected Tumblr palette; in an ideal world this would have no visual effect on any of these pages when an updated extension palette is selected. This isn't actually the case, since some components (premium perks button, a selected community in the sidebar, the text below the new post bar, etc) are coded with DOM changes based on the currently selected palette, but there should be fewer of these changes with this PR.
